app/uk/gov/hmrc/bindingtariffclassification/model/Event.scala

```diff
@@ -53,11 +59,34 @@ case class CaseStatusChange
   override val from: CaseStatus,
   override val to: CaseStatus,
   override val comment: Option[String] = None,
-  attachmentId: Option[String] = None
-) extends FieldChange[CaseStatus] {
+  override val attachmentId: Option[String] = None
+) extends FieldChange[CaseStatus] with OptionalAttachment {
   override val `type`: EventType.Value = EventType.CASE_STATUS_CHANGE
 }
 
+case class CancellationCaseStatusChange
+(
+  override val from: CaseStatus,
+  override val comment: Option[String] = None,
+  override val attachmentId: Option[String] = None,
+  reason: CancelReason
+) extends FieldChange[CaseStatus] with OptionalAttachment {
+  override val to: CaseStatus = CaseStatus.CANCELLED
+  override val `type`: EventType.Value = EventType.CASE_CANCELLATION
+}
+
+case class ReferralCaseStatusChange
+(
+  override val from: CaseStatus,
+  override val comment: Option[String] = None,
+  override val attachmentId: Option[String] = None,
+  subject: String,
```

Comment:
Is 'subject' = "who are you referring case to"?

Reply:
Yeh, feel free to suggest a more logical name, that was the best I could come up with

Reply:
I considered referee, but thats something else...

Reply:
"referredTo"?

Reply:
That will do